src/pad.ts

```diff
@@ -281,68 +279,78 @@ UI.pad.notepadToHTML = function (pad, kb) {
   return html
 }
 
-UI.pad.notepad = function (dom, padDoc, subject, me, options) {
+export function notepad (dom, padDoc, subject, me, options) {
   options = options || {}
-  var exists = options.exists
-  var table = dom.createElement('table')
-  var kb = UI.store
-  var ns = UI.ns
+  const exists = options.exists
+  const table = dom.createElement('table')
+  const kb = store
 
   if (me && !me.uri) throw new Error('UI.pad.notepad:  Invalid userid')
 
-  var updater = UI.store.updater
+  const updater = store.updater
 
-  var PAD = $rdf.Namespace('http://www.w3.org/ns/pim/pad#')
+  const PAD = Namespace('http://www.w3.org/ns/pim/pad#')
 
   table.setAttribute(
     'style',
     'padding: 1em; overflow: auto; resize: horizontal; min-width: 40em;'
   )
 
-  var upstreamStatus = null
-  var downstreamStatus = null
+  let upstreamStatus: HTMLElement | null = null
+  let downstreamStatus: HTMLElement | null = null
 
   if (options.statusArea) {
-    var t = options.statusArea.appendChild(dom.createElement('table'))
-    var tr = t.appendChild(dom.createElement('tr'))
+    const t = options.statusArea.appendChild(dom.createElement('table'))
+    const tr = t.appendChild(dom.createElement('tr'))
     upstreamStatus = tr.appendChild(dom.createElement('td'))
     downstreamStatus = tr.appendChild(dom.createElement('td'))
-    upstreamStatus.setAttribute('style', 'width:50%')
-    downstreamStatus.setAttribute('style', 'width:50%')
+    if (upstreamStatus) {
+      upstreamStatus.setAttribute('style', 'width:50%')
+    }
+    if (downstreamStatus) {
+      downstreamStatus.setAttribute('style', 'width:50%')
+    }
   }
-
-  var complain = function (message, upstream) {
+  /* @@ TODO want to look into this, it seems upstream should be a boolean and default to false ?
+  *
+  */
+  const complain = function (message: string, upstream: boolean = false) {
     console.log(message)
     if (options.statusArea) {
-      ;(upstream ? upstreamStatus : downstreamStatus).appendChild(
-        UI.widgets.errorMessageBlock(dom, message, 'pink')
-      )
+      if (upstream && upstreamStatus) {
```

Comment:
wanted to get this checked as there was an error the old way because it said upstreamStatus and downstreamStatus could both be null.

Reply:
The code assumes that the result of `tr.appendChild(dom.createElement('td'))` is not null, and I think that's a safe assumption, so you can just cast `(upstream ? upstreamStatus : downstreamStatus)` from HTMLElement|null to HTMLElement.

Reply:
The original code initialized these variables to null.  That's what made me have to do HTMLElement | null.  If I take out the initialization to null this does work though.  Do you think I should go this route (take out the initialization and leave in the upstream ? code)?

Reply:
No, you can leave the initialization to null and leave the variable type as `HTMLElement | null` but then cast to `HTMLElement` inside the if statement from line 319, because inside that if you know that lines 303-312 will have been executed earlier.

Reply:
will do.